src/RouteResolver.tsx

```diff
@@ -1,107 +1,150 @@
+          )}
+          <SynapseComponent
+            synapseConfig={props.config}
+            searchParams={searchParamsProps}
+          />
+        </div>
+      ) : (
+        <Layout
+          key={JSON.stringify(props)}
+          containerClassName={containerClassName}
+          outsideContainerClassName={outsideContainerClassName}
+        >
+          {scrollToJsx}
+          {/* re-think how this renders! remove specific styling */}
+          {title && (
+            <h2 className={`title ${centerTitle ? 'center-title' : ''}`}>
+              {title}
+            </h2>
+          )}
+          {subtitle && (
+            <p className={`${centerTitle ? 'center-title' : ''}`}>{subtitle}</p>
+          )}
+          <SynapseComponent
+            synapseConfig={props.config}
+            searchParams={searchParamsProps}
+          />
+        </Layout>
+      )}
+    </React.Fragment>
+  )
 }
 
-type SynapseComponentProps = {
-  synapseConfig: SynapseConfig
-}
+function RecursiveRouteRenderer(props: {
+  route: GenericRoute
+  getPageNameFromParentRoute: () => string
+}) {
+  const { route, getPageNameFromParentRoute } = props
+  const { url } = useRouteMatch()
 
-export const SynapseComponent: React.FC<SynapseComponentProps> = ({
-  synapseConfig,
-}) => {
-  const Component =
-    (PortalComponents as any)[synapseConfig.name] ??
-    (SynapseComponents as any)[synapseConfig.name]
-  if (!Component) {
-    throw Error(`No synapse object could be mapped for ${synapseConfig.name}`)
-  }
-  const component = <Component {...synapseConfig.props} />
-  const { style, className } = synapseConfig
-  if (style || className) {
-    return (
-      <div className={className} style={style}>
-        {component}
-      </div>
-    )
-  } else {
-    return component
+  const pageName = route.displayName ?? route.path?.replaceAll('/', '')
+  const getPageName = () =>
+    pageName
+      ? `${docTitleConfig.name} - ${pageName}`
+      : getPageNameFromParentRoute()
+
+  // if there are children, don't update the title
+  if (!route.routes || route.routes.length === 0) {
+    const newTitle: string = getPageName()
+    if (document.title !== newTitle) {
+      document.title = newTitle
+    }
   }
+
+  return (
+    <>
+      {'synapseConfigArray' in route &&
+        route.synapseConfigArray &&
+        route.synapseConfigArray.map((config, index) => {
+          return <ComponentRenderer key={index} config={config} />
+        })}
+      {'routes' in route &&
+        route.routes &&
+        route.routes.map((r) => {
+          return (
+            <Route
+              key={JSON.stringify(r)}
+              path={`${url}${url.endsWith('/') ? '' : '/'}${r.path}`}
+              exact={r.exact}
+            >
+              <RecursiveRouteRenderer
+                route={r}
+                getPageNameFromParentRoute={getPageName}
+              />
+            </Route>
+          )
+        })}
+    </>
+  )
```

Comment:
If this "node" has components, render them.

If it has children, render them recursively within a Route.